src/hardware/hantek-dso/protocol.c

```diff
@@ -241,7 +241,187 @@ static int get_channel_offsets(const struct sr_dev_inst *sdi)
 	return SR_OK;
 }
 
-static int dso_set_trigger_samplerate(const struct sr_dev_inst *sdi)
+
+static void dso2250_set_triggerpos(int value, int long_buffer, uint8_t dest[], int offset)
+{
+	uint32_t min, max;
+	uint32_t tmp, diff;
+
+	min = long_buffer ? 0 : 0x7d7ff;
+	max = 0x7ffff;
+
+	diff = max - min;
+	tmp = min + diff * value / 100;
+	sr_dbg("2250 trigger pos: %3d%% * [0x%x,0x%x] == 0x%x", value, min, max, tmp);
+
+	dest[offset + 0] = tmp & 0xff;
+	dest[offset + 1] = (tmp >> 8) & 0xff;
+	dest[offset + 2] = (tmp >> 16) & 0x7;
+}
+
+
+/* See http://openhantek.sourceforge.net/doc/namespaceHantek.html#ac1cd181814cf3da74771c29800b39028 */
+static int dso2250_set_trigger_samplerate(const struct sr_dev_inst *sdi)
+{
+	struct dev_context *devc;
+	struct sr_usb_dev_inst *usb;
+	int ret, tmp;
+	int base;
+	uint8_t cmdstring[12];
+	int trig;
+
+
+	sr_dbg("Preparing CMD_2250_SET_TRIGGERSOURCE.");
+
+	devc = sdi->priv;
+	usb = sdi->conn;
+
+	memset(cmdstring, 0, sizeof(cmdstring));
+	/* Command */
+	cmdstring[0] = CMD_2250_SET_TRIGGERSOURCE;
+	sr_dbg("Trigger source %s.", devc->triggersource);
+	if (!strcmp("CH2", devc->triggersource))
+		tmp = 3;
+	else if (!strcmp("CH1", devc->triggersource))
+		tmp = 2;
+	else if (!strcmp("EXT", devc->triggersource))
+		tmp = 0;
+	else {
+		sr_err("Invalid trigger source: '%s'.", devc->triggersource);
```

Comment:
this doesn't log at a higher level when it gets the SR_ERR_ARG?

Reply:
No (not yet). Most of that is still left over from the non-2250 code that I copied...